homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -169,24 +177,40 @@ async def async_setup_entry(
     _LOGGER.debug("Connected to Lutron Caseta bridge via LEAP at %s", host)
     await _async_migrate_unique_ids(hass, config_entry)
 
-    devices = bridge.get_devices()
-    bridge_device = devices[BRIDGE_DEVICE_ID]
+    bridge_devices = bridge.get_devices()
+    bridge_device = bridge_devices[BRIDGE_DEVICE_ID]
+
     if not config_entry.unique_id:
         hass.config_entries.async_update_entry(
             config_entry, unique_id=serial_to_unique_id(bridge_device["serial"])
         )
 
-    buttons = bridge.buttons
     _async_register_bridge_device(hass, entry_id, bridge_device, bridge)
-    button_devices, device_info_by_device_id = _async_register_button_devices(
-        hass, entry_id, bridge, bridge_device, buttons
-    )
-    _async_subscribe_pico_remote_events(hass, bridge, buttons)
+
+    (
+        keypads,
+        keypad_buttons,
+        keypad_button_names_to_leap,
+        dr_device_id_to_keypad,
+    ) = _async_register_keypads(hass, entry_id, bridge, bridge_device)
+
+    keypad_trigger_schemas = _async_build_trigger_schemas(keypad_button_names_to_leap)
```

Comment:
Maybe rename `_async_register_keypads` to `_async_setup_keypads` and have it return `LutronKeypadData` and make it responsible for calling `_async_build_trigger_schemas` and `_async_subscribe_keypad_events`

Then you can avoid the 4 item tuple return.

Reply:
done.  Great idea.  Much Cleaner.